### PR TITLE
fix(compiler): improve error message for references to default exports

### LINKED ISSUE #11034: bug(compiler) DI token for third-party-library crashes the AOT compiler
**I'm submitting a ...**  (check one with "x")

```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

Trying to provide a token for a a third-party-library object lets the AOT compiler crash. Having a Module like this:

``` ts
import * as io from 'socket.io-client';

@NgModule({
  providers: [
    ...
    {provide: "SocketIO", useValue: io},
  ],
  ...
})
export class AppModule {
}
```

calling

```
./node_modules/.bin/ngc -p ./src
```

throws the following exception:

```
Error: Internal error: unknown identifier [object Object]
    at new BaseException (/home/christoph/buch/sources/project-manager-reactive-2/node_modules/@angular/core/src/facade/exceptions.js:27:23)
    at _TsEmitterVisitor._visitIdentifier (/home/christoph/buch/sources/project-manager-reactive-2/node_modules/@angular/compiler/src/output/ts_emitter.js:293:19)
    at _TsEmitterVisitor.visitExternalExpr (/home/christoph/buch/sources/project-manager-reactive-2/node_modules/@angular/compiler/src/output/ts_emitter.js:84:14)
    at ExternalExpr.visitExpression (/home/christoph/buch/sources/project-manager-reactive-2/node_modules/@angular/compiler/src/output/output_ast.js:348:24)
    at _TsEmitterVisitor.AbstractEmitterVisitor.visitWritePropExpr (/home/christoph/buch/sources/project-manager-reactive-2/node_modules/@angular/compiler/src/output/abstract_emitter.js:185:20)
    at WritePropExpr.visitExpression (/home/christoph/buch/sources/project-manager-reactive-2/node_modules/@angular/compiler/src/output/output_ast.js:265:24)
    at _TsEmitterVisitor.AbstractEmitterVisitor.visitExpressionStmt (/home/christoph/buch/sources/project-manager-reactive-2/node_modules/@angular/compiler/src/output/abstract_emitter.js:104:19)
    at ExpressionStatement.visitStatement (/home/christoph/buch/sources/project-manager-reactive-2/node_modules/@angular/compiler/src/output/output_ast.js:562:24)
    at /home/christoph/buch/sources/project-manager-reactive-2/node_modules/@angular/compiler/src/output/abstract_emitter.js:384:58
    at Array.forEach (native)
c
```

after some research it looks like the aot compiler is iterating over all tokens and tries to precompile them, but that doesn't work for the provided function

**Expected/desired behavior**

it should be possible to provide functions from third-party-libraries when using the AOT compiler

**Please tell us about your environment:**
- **Angular version:** commit 9883e19e2e2839043557ae89740a8f6b299680d2
- **Browser:** [all  ] 
- **Language:** [all ]

### LINKED ISSUE #12020: Cryptic error message when using default exports.
**I'm submitting a ...**  (check one with "x")

```
[x] bug report => search github for a similar issue or PR before submitting
```

**Current behavior** 

The project https://github.com/beginor/learning-angular2 produces an error message "Cannot read property 'type' of 'null'". 

From a Stack Overflow question:
http://stackoverflow.com/questions/39190493/get-cannot-read-property-type-of-null-using-angular2-aot-ngc-compiler-with-r

**Expected behavior**

The compiler should report a message detailing the unsupported feature or produce a working program.

**Reproduction of the problem**

Execute `ngc` on the project in https://github.com/beginor/learning-angular2

### PR DESCRIPTION
**Pull request by chuckjaz**
Fixes #12020
Fixes #11034

**What kind of change does this PR introduce?** (check one with "x")
```
[x] Bugfix
```

**What is the current behavior?** (You can also link to an open issue here)

The error message produced when references to module instances and default exports are cryptic.

**What is the new behavior?**

The error messages are more descriptive of the actual error.

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: modules/@angular/compiler/test/aot/static_symbol_resolver_spec.ts [hunk 1]
```diff
@@ -235,16 +235,20 @@ describe('StaticSymbolResolver', () => {
     });
   });
 
-  it('should ignore module references without a name', () => {
+  it('should report an error for references to defautl exports', () => {
```
Comment (alfaproject) [inline modules/@angular/compiler/test/aot/static_symbol_resolver_spec.ts]: default

Comment (chuckjaz) [inline modules/@angular/compiler/test/aot/static_symbol_resolver_spec.ts]: Thanks!

File: modules/@angular/compiler/test/aot/compiler_spec.ts [hunk 2]
```diff
@@ -67,6 +67,69 @@ describe('compiler (unbundled Angular)', () => {
          expect(generatedFiles.find(f => /app\.module\.ngfactory\.ts/.test(f.genFileUrl)))
              .toBeDefined();
        })));
+
+    it('should be report references to default exports', async(() => {
```
Comment (tbosch) [inline modules/@angular/compiler/test/aot/compiler_spec.ts]: typo: `be report`...

**Review by tbosch — APPROVED** [review]

**Comment by chuckjaz** [comment]
Obsolete

**Pull request closed without merging**